File: shoring/blackjackEngine/src/main/java/com/panopset/blackjackEngine/DefaultResources.java
package com.panopset.blackjackEngine;

import com.panopset.compat.Streamop;

public enum DefaultResources {
  
  INSTANCE;

  private static final String DEFAULT_STRATEGY_RSRC

      = "/basic.txt";

  private static final String DEFAULT_COUNTING_SYSTEMS_RSRC

      = "/cs.txt";

  public static String getDefaultBasicStrategy() {
    return INSTANCE.getDefaultBasicStrategyPrivate();
  }
  
  private String getDefaultBasicStrategyPrivate() {
    return Streamop.getTextFromStream(this.getClass().getResourceAsStream(DEFAULT_STRATEGY_RSRC));
  }

  public static String getDefaultCountingSystems() {
    return INSTANCE.getDefaultCountingSystemsPrivate();
  }
  
  private String getDefaultCountingSystemsPrivate() {
    return Streamop.getTextFromStream(this.getClass().getResourceAsStream(DEFAULT_COUNTING_SYSTEMS_RSRC));
  }
}

File: code/shoring/blackjackEngine/src/main/kotlin/com/panopset/blackjackEngine/DefaultResources.kt
package com.panopset.blackjackEngine

import com.panopset.compat.Streamop.getTextFromStream

class DefaultResources() {
    val defaultBasicStrategy = getTextFromStream(this.javaClass.getResourceAsStream(DEFAULT_STRATEGY_RSRC))
    val defaultCountingSystems = getTextFromStream(this.javaClass.getResourceAsStream(DEFAULT_COUNTING_SYSTEMS_RSRC))
}

private const val DEFAULT_STRATEGY_RSRC = "/basic.txt"
private const val DEFAULT_COUNTING_SYSTEMS_RSRC = "/cs.txt"
